### DISCUSSION TITLE
Vague Proposal: how do we get this pushed upstream?

### DISCUSSION THREAD

**Discussion by bkeane** [awslabs/aws-lambda-web-adapter#549] [General]
Lambda web adapter allows me to better achieve 95% of my end goals. 

I believe lambda as a service should _allow_ for the lower level interface we know and love as emulated by [this guy](https://github.com/aws/aws-lambda-runtime-interface-emulator).

This web adapter as an interface design for Lambda itself, however, is so remarkably useful that I'd prefer the Lambda service to actually just provide this as an "executor" option or the like.

If anyone from the AWS Lambda product team is on here, or knows how to pull some attention, here is what I would state from a product advocacy standpoint:

I am comfortable with the lowest layers of lambda, but this is what I use. It is _the right choice_ for such a vast number of tasks in AWS Lambda, that I honestly can't understand what attracted me to Lambda without it.

I love that I can dive below the abstraction layer this provides for data pipelines etc. But even then? I find myself just reaching for this and leave dropping to standard invoke as a "premature optimization" I can easily mutate my code to. Not sayable for the other direction.

It would be really nifty if I could exclude this plugin from my docker images, making them 100% agnostic of runtime environment, leaving only a web server.

This is where I point people to learn how to use Lambda. This is also where I advise tenured folks to gravitate for the sake of... everything.

It's a shot in the dark, but I really think AWS should take this thing into the fold. It would just make the product better out-of-the-box.

**Comment by bnusunny** [comment]
Thanks for this suggestion. I will pass it to the Lambda team. Please keep such feedback coming. :D
